test/common/filter/reader/example/filter_integration_test.cc

```diff
@@ -2,16 +2,16 @@
 #include "test/integration/utility.h"
 
 namespace Envoy {
-class Echo2IntegrationTest : public BaseIntegrationTest,
-                             public testing::TestWithParam<Network::Address::IpVersion> {
+class ExampleReaderIntegrationTest : public BaseIntegrationTest,
```

Comment:
Do you think it's worth reorganizing the HTTP filter as well in this repo-wide restructure?

Reply:
Yes, my intention actually to collect examples to make it ready for answering issues. For example something like https://github.com/envoyproxy/envoy/issues/2635.

Reply:
I will put the http ones in this PR.

Reply:
@htuch the current layout:

```
|-- source
|   |-- BUILD
|   `-- extensions
|       `-- filters
|           |-- http
|           |   `-- example
|           |       |-- README.md
|           |       |-- decoder
|           |       |   |-- BUILD
|           |       |   |-- README.md
|           |       |   |-- filter.cc
|           |       |   |-- filter.h
|           |       |   |-- filter.proto
|           |       |   `-- filter_config.cc
|           |       |-- decoder_encoder
|           |       |   |-- BUILD
|           |       |   |-- README.md
|           |       |   |-- filter.cc
|           |       |   |-- filter.h
|           |       |   |-- filter.proto
|           |       |   `-- filter_config.cc
|           |       `-- encoder
|           |           |-- BUILD
|           |           |-- README.md
|           |           |-- filter.cc
|           |           |-- filter.h
|           |           |-- filter.proto
|           |           `-- filter_config.cc
|           `-- network
|               `-- example
|                   |-- README.md
|                   `-- reader
|                       |-- BUILD
|                       |-- README.md
|                       |-- filter.cc
|                       |-- filter.h
|                       |-- filter.proto
|                       `-- filter_config.cc
|-- test
|   |-- BUILD
|   |-- envoy_binary_test.sh
|   |-- extensions
|   |   `-- filters
|   |       |-- http
|   |       |   `-- example
|   |       |       |-- decoder
|   |       |       |   `-- filter_test.cc
|   |       |       |-- decoder_encoder
|   |       |       |   `-- filter_test.cc
|   |       |       `-- encoder
|   |       |           `-- filter_test.cc
|   |       `-- network
|   |           `-- example
|   |               `-- reader
|   |                   `-- filter_test.cc
|   |-- fixtures
|   |   `-- configs
|   |       |-- BUILD
|   |       `-- v2
|   |           `-- echo.yaml
|   `-- integration
|       `-- extensions
|           `-- filters
|               |-- http
|               |   |-- BUILD
|               |   |-- example_decoder_encoder_test.cc
|               |   |-- example_decoder_test.cc
|               |   `-- example_encoder_test.cc
|               `-- network
|                   |-- BUILD
|                   `-- example_reader_test.cc
```